compiler/packages/babel-plugin-react-compiler/src/Optimization/InstructionReordering.ts

```diff
@@ -319,8 +379,7 @@ function getDepth(env: Environment, nodes: Nodes, id: IdentifierId): number {
     return node.depth;
   }
   node.depth = 0; // in case of cycles
-  let depth =
-    node.instruction != null && mayAllocate(env, node.instruction) ? 1 : 0;
+  let depth = node.reorderability === Reorderability.Reorderable ? 0 : 1;
```

Comment:
in earlier versions the depth calculation was wrong, we don't want to align to mayAllocate() but rather count non-reorderable instruction as having higher priority than reorderable ones.